besu/src/main/java/org/hyperledger/besu/controller/QbftBesuControllerBuilder.java

```diff
@@ -262,11 +275,20 @@ protected BftContext createConsensusContext(
     final Map<Long, List<Address>> bftValidatorForkMap =
         convertBftForks(configOptions.getTransitions().getQbftForks());
 
-    return new BftContext(
-        BlockValidatorProvider.forkingValidatorProvider(
-            blockchain, epochManager, bftBlockInterface().get(), bftValidatorForkMap),
-        epochManager,
-        bftBlockInterface().get());
+    if (pkiKeyStore.isPresent()) {
+      return new PkiQbftContext(
+          BlockValidatorProvider.forkingValidatorProvider(
```

Comment:
Would be nice to move the creation of the validatorProvider out of the if/else into it's own variable